lib/client/client.go

```diff
@@ -375,7 +375,7 @@ func (proxy *ProxyClient) IssueUserCertsWithMFA(ctx context.Context, params Reis
 	defer stream.CloseSend()
 
 	initReq := &proto.UserCertsRequest{
-		Username:          proxy.hostLogin,
+		Username:          cert.KeyId,
```

Comment:
Not related to your change, just noticing that `RouteToApp` is not set in this request for some reason.

Reply:
https://github.com/gravitational/teleport/pull/5938 will end up setting it